src/main/java/kafdrop/service/KafkaHighLevelConsumer.java

```diff
@@ -204,11 +204,19 @@ synchronized List<ConsumerRecord<String, String>> getLatestRecords(String topic,
    */
   synchronized SearchResults searchRecords(String topic,
                                            String searchString,
+                                           Integer partition,
                                            Integer maximumCount,
                                            Date startTimestamp,
                                            Deserializers deserializers) {
     initializeClient();
-    final List<TopicPartition> partitions = determinePartitionsForTopic(topic);
+    List<TopicPartition> partitions = determinePartitionsForTopic(topic);
+    if (partition != -1) {
+      var partitionOpt = partitions.stream().filter(p -> p.partition() == partition).findAny();
+      if (partitionOpt.isEmpty()) {
+        throw new IllegalArgumentException("Partition does not exist in topic");
+      }
+      partitions = List.of(partitionOpt.get());
```

Comment:
Make this a little more compact:
```suggestion
      partitions = List.of(partitionOpt.orElseThrow(() -> new IllegalArgumentException("Partition " + partition + " does not exist in topic")));
 ```

Reply:
Thank you sir